### PR TITLE
fix wrong effect of a circular path payment

### LINKED ISSUE #744: wrong effects for circular path payments
Path payment operations that start and end with the same asset and account seem to report both `account_debited` and `account_credited` to be the credited amount.

example: 
operation https://horizon.stellar.org/operations/89293654279069697/
```
"source_amount": "0.1081398",
"amount": "0.1100200",
```
but effects https://horizon.stellar.org/operations/89293654279069697/effects
```
 "type": "account_credited",
 "amount": "0.1100200"
...
"type": "account_debited",
"amount": "0.1100200"
```

### PR DESCRIPTION
**Pull request by howardtw**
`account_debited` should indicate the `source_amount` instead.

Close #744

### CODE REVIEW

**Comment by bartekn** [comment]
This looks good to me but after checking `PathPaymentResult.SendAmount()` I suspect there's a bug. Looks like it checks all consumed offers and keeps adding `AmountBought` to value that is later returned. The problem is if source asset appears more than one time on the path it will be incorrect value. Let me confirm this in testnet.

**Comment by howardtw** [comment]
@bartekn added a check in the latest commit. But I still need to know how to update `kahuna-horizon.sql` :)

File: services/horizon/internal/ingest/session_test.go [hunk 1]
```diff
@@ -74,6 +74,16 @@ func Test_ingestOperationEffects(t *testing.T) {
...
+	var details struct {
+		Amount float64 `json:"amount,string"`
+	}
+	err = effects[1].UnmarshalDetails(&details)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if details.Amount != 100 {
+		t.Errorf("got: %v, want: 100, details: %s", details.Amount, effects[1].DetailsString.String)
+	}
```
Comment (bartekn) [inline services/horizon/internal/ingest/session_test.go]: You don't need to unmarshal it manually, we already have methods to do it. Check this:
```diff
diff --git a/services/horizon/internal/ingest/session_test.go b/services/horizon/internal/ingest/session_test.go
index 4223cea..af495fd 100644
--- a/services/horizon/internal/ingest/session_test.go
+++ b/services/horizon/internal/ingest/session_test.go
@@ -3,6 +3,7 @@ package ingest
 import (
        "testing"
 
+       protocolEffects "github.com/stellar/go/protocols/horizon/effects"
        "github.com/stellar/go/services/horizon/internal/db2"
        "github.com/stellar/go/services/horizon/internal/db2/history"
        "github.com/stellar/go/services/horizon/internal/test"
@@ -72,17 +73,11 @@ func Test_ingestOperationEffects(t *testing.T) {
                tt.Assert.Equal(history.EffectAccountDebited, effects[1].Type)
                tt.Assert.Equal(history.EffectTrade, effects[2].Type)
                tt.Assert.Equal(history.EffectTrade, effects[3].Type)
-       }
 
-       var details struct {
-               Amount float64 `json:"amount,string"`
-       }
-       err = effects[1].UnmarshalDetails(&details)
-       if err != nil {
-               t.Fatal(err)
-       }
-       if details.Amount != 100 {
-               t.Errorf("got: %v, want: 100, details: %s", details.Amount, effects[1].DetailsString.String)
+               e := protocolEffects.AccountDebited{}
+               err = effects[1].UnmarshalDetails(&e)
+               tt.Require.NoError(err)
+               tt.Assert.Equal("100.0000000", e.Amount)
        }
 }
 ```
The problem is that it's not testing correct scenario. I reverted `session.go` to `master` version and the test passed too. You can get effects for the affected operation by using something like:
```go
err = q.Effects().ForOperation(81604382721).Page(pq).Select(&effects)
```

Comment (howardtw) [reply services/horizon/internal/ingest/session_test.go]: Oh I guess I don't have to declare an anonymous struct, but I'm already using the method you suggested. 😄 Ok great! I'll use `ForOperation` instead

**Comment by howardtw** [comment]
@bartekn PTAL again. I updated the test to be using the operation number as you suggested. I also ran it against master, and it failed as expected.

I also ran the script that generates the scenario sql files, and it produced a lot of diffs. I think it's due to the fast that I'm using Postgres 10.4 whereas the sql files in the repo were dumped by Postgres 9.6.1. All the table names in my dump has the prefix of `public`. I wonder if it's expected. Having said that, I think it might be better to separate the dump commit to a standalone PR.

**Comment by howardtw** [comment]
@bartekn incremented `CurrentVersion` in aeb4fa2b6de401352e8418cf251675b97c34f4f9

**Pull request merged**